tests/draft2020-12/default.json

```
@@ -45,5 +45,40 @@
                 "valid": true
             }
         ]
+    },
+    {
+        "description": "the default keyword does not do anything if the property is missing",
+        "schema": {
+            "type": "object",
+            "properties": {
+                "alpha": {
+                    "type": "number",
+                    "maximum": 5,
+                    "default": 1
+                }
+            },
+            "patternProperties": {
+                "a": {
+                    "minimum": 3
+                }
+            }
+        },
+        "tests": [
+            {
+                "description": "an explicit property value is checked against both keywords (passing)",
+                "data": { "alpha": 4 },
+                "valid": true
+            },
+            {
+                "description": "an explicit property value is checked against both keywords (failing)",
+                "data": { "alpha": 1 },
+                "valid": false
+            },
+            {
+                "description": "missing properties are not filled in with the default",
+                "data": {},
+                "valid": true
+            }
+        ]
```

This seems more complicated than it needs to be. What is the purpose of including `patternProperties`? Here's what I would suggest.

```suggestion
        "description": "the default keyword does not do anything if the property is missing",
        "schema": {
            "type": "object",
            "properties": {
                "alpha": { "default": 1 }
            },
            "required": ["alpha"]
        },
        "tests": [
            {
                "description": "an explicit property exists",
                "data": { "alpha": 4 },
                "valid": true
            },
            {
                "description": "missing properties are not filled in with the default",
                "data": {},
                "valid": false
            }
        ]
```

---

I used 'maximum' as the test rather than 'required' because I have seen some implementations treat other object-based keywords differently than others. On reflection, I think I would use both, and drop patternProperties (I can't remember what I had in mind for that one).